### ISSUE TITLE
Add an analyzer to prevent indexing a localized value instead of member name

### ISSUE THREAD

**Issue by adamint** [dotnet/aspire#6352]
See https://github.com/dotnet/aspire/pull/6209#discussion_r1803940775

We've had oodles of this kind of issue already, and it is incredibly easy to forget to add `nameof` to an `IStringLocalizer` call. We should create an analyzer to prevent it in the future.

Right:
`IStringLocalizer<T>[nameof(T.Field)]`

Wrong:
`IStringLocalizer<T>[T.Field]`

**Comment by adamint** [comment]
@drewnoakes This analyzer would be appropriate in ASP.NET because it could be used for all applications - https://github.com/dotnet/aspnetcore/tree/main/src/Analyzers

Should it go into the aspire repo first, or the above location?

**Comment by drewnoakes** [comment]
Yep, ASP.NET Core owns [`Microsoft.Extensions.Localization.IStringLocalizer`](https://github.com/dotnet/aspnetcore/blob/21e8836755b8be5f7fc7bf35d9f5b87ae60b62a0/src/Localization/Abstractions/src/IStringLocalizer.cs).

Any analyzer for this _could_ be added to [`Microsoft.AspNetCore.Analyzers`](https://github.com/dotnet/aspnetcore/tree/main/src/Analyzers/Analyzers/src), but there may be a better place for `MS.Extensions` analyzers (or maybe one should be made).

Either way this issue can be migrated to ASP.NET Core.

**Comment by drewnoakes** [comment]
Bugs like those fixed in https://github.com/dotnet/aspire/pull/6348 are motivation for this analyzer.

**Comment by adamint** [comment]
Should I transfer the issue? or should we implement

**Comment by drewnoakes** [comment]
Transfer

**Comment by joperezr** [comment]
[Triage] Discussed this during triage and seems like the localization feature here isn't used as it was designed to. You could probably look into po files. Let's keep this open to track the work of using localizable strings correctly in the dashboard as per @drewnoakes request.

**Comment by adamint** [comment]
> [Triage] Discussed this during triage and seems like the localization feature here isn't used as it was designed to. You could probably look into po files. Let's keep this open to track the work of using localizable strings correctly in the dashboard as per [@drewnoakes](https://github.com/drewnoakes) request.

I’m confused, does our tooling even support the use of PO files?

**Comment by joperezr** [comment]
Consider moving this to aspnetcore repo.
